Libraries/Text/RCTTextField.m

```diff
@@ -173,16 +192,19 @@ - (void)textFieldSubmitEditing
 
 - (void)textFieldBeginEditing
 {
-  if (_selectTextOnFocus) {
-    dispatch_async(dispatch_get_main_queue(), ^{
-      [self selectAll:nil];
-    });
-  }
   [_eventDispatcher sendTextEventWithType:RCTTextEventTypeFocus
                                  reactTag:self.reactTag
                                      text:self.text
                                       key:nil
                                eventCount:_nativeEventCount];
+
+  dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
I don't think it should cause issues but I'll do some more test that to make sure. I think it will change the UITextField selection when calling selectAll then sendSelectionEvent should trigger onSelectionChange with the text fully selected. At the point JS can handle this like any other text selection change made by the user.